src/common/mfu_flist_walk.c

```diff
@@ -116,6 +117,7 @@ static int build_path(char* path, size_t path_len, const char* dir, const char*
     if (new_len > path_len) {
         MFU_LOG(MFU_LOG_ERR, "Path name is too long, %lu chars exceeds limit %lu: '%s/%s'",
                 new_len, path_len, dir, name);
+        WALK_RESULT = errno;
```

Comment:
oops, good catch.
Since WALK_RESULT is just one int, it's not of much use to store errno anyway, as there may be many different errors in a single walk.  I'll switch to storing -1 to be consistent with similar code elsewhere in the project.